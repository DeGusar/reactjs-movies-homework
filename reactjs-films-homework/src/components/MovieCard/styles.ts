import { makeStyles } from "@mui/styles";

type PropsType = {
  imageHeight: string
}

export const useStyles = makeStyles ((props: PropsType) => ({
  cardWrapper: (props: PropsType) => ({
    height: `${props.imageHeight}`,
    position: 'relative',
    backgroundColor: 'rgb(232,232,232)',
    cursor: 'pointer'
  }),

  cardImage: {
    objectFit: 'cover',
    width: '100%',
    height: '100%'
  },

  lowRating: {
    position: 'absolute',
    width: '40px',
    height: '22px',
    top: '10px',
    left: '-10px',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    color: 'white',
    backgroundColor: 'rgb(170,170,170)'
  },

  highRating: {
    position: 'absolute',
    width: '40px',
    height: '22px',
    top: '10px',
    left: '-10px',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    color: 'white',
    backgroundColor: 'rgb(42,125,40)'
  },

  title: {
    paddingTop: '10px',
    color: 'rgb(185,185,185)'
  },
  genre: {
    paddingTop: '5px',
    color: 'rgb(185,185,185)'
  }
}))