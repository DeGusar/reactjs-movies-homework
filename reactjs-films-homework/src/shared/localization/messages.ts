import enMessages from './en.json';
import ruMessages from './ru.json';

type Messages = {
  [key: string]: { [key: string]: string };
};

const messages: Messages = {
  en: enMessages as { [key: string]: string } ,
  ru: ruMessages as { [key: string]: string },
};

export default messages;
