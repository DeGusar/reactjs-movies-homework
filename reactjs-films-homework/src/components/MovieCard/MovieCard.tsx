import { Box, Paper, Stack, Typography } from '@mui/material';
import { useNavigate } from 'react-router-dom';
import { useStyles } from './styles';

type MovieCardProps = {
  popularity: string;
  id: number;
  imageHeight: string;
  isDescription?: boolean;
};

export const MovieCard = ({
  popularity,
  id,
  imageHeight,
  isDescription = true,
}: MovieCardProps) => {
  const classes = useStyles({ imageHeight });
  const navigate = useNavigate();

  const handleClick = () => navigate(`/movie/${id}`);

  return (
    <>
      <Box className={classes.cardWrapper} onClick={handleClick}>
        <Box
          component="img"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/328.jpg"
          className={classes.cardImage}
        ></Box>
        <Paper className={Number(popularity) >= 7 ? classes.highRating : classes.lowRating}>
          {popularity}
        </Paper>
      </Box>
      {isDescription && (
        <>
          <Typography className={classes.title}>MovieTitle</Typography>
          <Stack>
            <Typography className={classes.genre}>Genre</Typography>
          </Stack>
        </>
      )}
    </>
  );
};
