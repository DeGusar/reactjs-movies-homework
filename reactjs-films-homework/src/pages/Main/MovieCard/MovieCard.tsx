import { Box, Paper, Stack, Typography } from '@mui/material';
import { useNavigate } from 'react-router-dom';
import { useStyles } from './styles';
import Image from 'material-ui-image';

type MovieCardProps = {
  popularity: String;
  id: Number;
};

export const MovieCard = ({ popularity, id }: MovieCardProps) => {
  const classes = useStyles();
  const navigate = useNavigate();

  const handleClick = () => navigate(`/person/${id}`);

  return (
    <>
      <Box className={classes.cardWrapper} onClick={handleClick}>
        <Box
          component="img"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/328.jpg"
          className={classes.cardImage}
        ></Box>
        <Paper className={Number(popularity) > 7 ? classes.highRating : classes.lowRating}>
          {popularity}
        </Paper>
      </Box>
      <Typography className={classes.title}>MovieTitle</Typography>
      <Stack>
        <Typography className={classes.genre}>Genre</Typography>
      </Stack>
    </>
  );
};
