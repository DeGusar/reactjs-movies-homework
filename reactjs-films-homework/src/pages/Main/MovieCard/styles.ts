import { makeStyles } from "@mui/styles";

export const useStyles = makeStyles (() => ({
  cardWrapper: {
    height: '350px',
    position: 'relative',
    backgroundColor: 'rgb(232,232,232)',
    cursor: 'pointer'
  },

  cardImage: {
    objectFit: 'cover',
    width: '100%',
    height: '100%'
  },

  lowRating: {
    position: 'absolute',
    width: '40px',
    height: '22px',
    top: '10px',
    left: '-10px',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    color: 'white',
    backgroundColor: 'rgb(170,170,170)'
  },

  highRating: {
    position: 'absolute',
    width: '40px',
    height: '22px',
    top: '10px',
    left: '-10px',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    color: 'white',
    backgroundColor: 'rgb(42,125,40)'
  },

  title: {
    paddingTop: '10px',
    color: 'rgb(185,185,185)'
  },
  genre: {
    paddingTop: '5px',
    color: 'rgb(185,185,185)'
  }
}))