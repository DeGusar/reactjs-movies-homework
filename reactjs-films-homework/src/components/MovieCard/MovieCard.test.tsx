import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MovieCard } from './MovieCard';
import { IntlProvider } from 'react-intl';
import { BrowserRouter } from 'react-router-dom';
import messages from '../../shared/localization/messages';

describe('MovieCard component', () => {
  const defaultProps = {
    popularity: '8',
    id: 1234,
    imageHeight: '400px',
    isShowDescription: true,
  };

  it('MovieCard component should render correctly', () => {
    render(
      <IntlProvider locale="en" messages={messages.en}>
        <BrowserRouter>
          <MovieCard {...defaultProps} />
        </BrowserRouter>
      </IntlProvider>
    );
    expect(screen.getByTestId('movie-card-title')).toBeInTheDocument();
    expect(screen.getByTestId('movie-card-rating')).toBeInTheDocument();
    expect(screen.getByTestId('movie-card-genre')).toBeInTheDocument();
  });

  it('MovieCard snapshot', () => {
    const view = render(
      <IntlProvider locale="en" messages={messages.en}>
        <BrowserRouter>
          <MovieCard {...defaultProps} />
        </BrowserRouter>
      </IntlProvider>
    );

    expect(view).toMatchSnapshot();
  });
});
