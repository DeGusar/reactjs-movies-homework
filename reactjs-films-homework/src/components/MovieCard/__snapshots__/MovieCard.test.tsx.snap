// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component MovieCard snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-cardWrapper-8 makeStyles-cardWrapper-14 MuiBox-root css-0"
      >
        <img
          class="makeStyles-cardImage-9 MuiBox-root css-0"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/328.jpg"
        />
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-highRating-11 css-1ps6pg7-MuiPaper-root"
          data-testid="movie-card-rating"
        >
          8
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 makeStyles-title-12 css-ahj2mt-MuiTypography-root"
        data-testid="movie-card-title"
      >
        MovieTitle
      </p>
      <div
        class="css-nen11g-MuiStack-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 makeStyles-genre-13 css-ahj2mt-MuiTypography-root"
          data-testid="movie-card-genre"
        >
          Genre
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-cardWrapper-8 makeStyles-cardWrapper-14 MuiBox-root css-0"
    >
      <img
        class="makeStyles-cardImage-9 MuiBox-root css-0"
        src="https://kinopoiskapiunofficial.tech/images/posters/kp/328.jpg"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-highRating-11 css-1ps6pg7-MuiPaper-root"
        data-testid="movie-card-rating"
      >
        8
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 makeStyles-title-12 css-ahj2mt-MuiTypography-root"
      data-testid="movie-card-title"
    >
      MovieTitle
    </p>
    <div
      class="css-nen11g-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 makeStyles-genre-13 css-ahj2mt-MuiTypography-root"
        data-testid="movie-card-genre"
      >
        Genre
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
